homeassistant/components/device_tracker/ubee.py

```diff
@@ -0,0 +1,127 @@
+"""
+Support for Ubee router.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.ubee/
+"""
+import logging
+import re
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+_ROUTER_REGEX = re.compile(
+    r'<tr bgcolor=#[0-9a-fA-F]+>'
+    r'<td>([0-9a-fA-F]{2}:[0-9a-fA-F]{2}:[0-9a-fA-F]{2}:[0-9a-fA-F]{2}:'
+    r'[0-9a-fA-F]{2}:[0-9a-fA-F]{2})</td>'
+    r'<td>\d+</td><td>.+</td><td>\d+\.\d+\.\d+\.\d+</td><td>(.+)</td>'
+    r'<td>.+</td><td>\d+</td></tr>'
+)
+_MAC_REGEX = re.compile(r'(([0-9A-Fa-f]{1,2}\:){5}[0-9A-Fa-f]{1,2})')
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return a Ubee scanner."""
+    try:
+        return UbeeRouterDeviceScanner(config[DOMAIN])
+    except ConnectionError:
+        return None
+
+
+class UbeeRouterDeviceScanner(DeviceScanner):
+    """This class queries a wireless Ubee router."""
+    def __init__(self, config):
+        """Initialize the Ubee scanner."""
+        self.host = config[CONF_HOST]
+        self.username = config[CONF_USERNAME]
+        self.password = config[CONF_PASSWORD]
+        self.login_url = 'http://{}/goform/login'.format(self.host)
+        self.status_url = 'http://{}/UbeeAdvConnectedDevicesList.asp'.format(
+            self.host
+        )
+
+        self.last_results = {}
+        self.mac2name = {}
+
+        # Test the router is accessible
+        payload = {
+            'loginUsername': self.username,
+            'loginPassword': self.password
+        }
+        try:
+            response = requests.post(self.login_url, data=payload, timeout=4)
```

Comment:
All communication should be extracted and handled in a separate package hosted on PyPI.
<https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services>

Reply:
Thanks @ludeeus.
I've updated the code to use a separate package `pyubee` hosted on PyPI.

Reply:
Hey @ludeeus , is there anything else I should do to get this merged? I've made changes you requested, all checks passed, documentation is in place.